src/components/calcite-input/calcite-input.tsx

```diff
@@ -244,6 +244,16 @@ export class CalciteInput {
 
   @State() localizedValue: string;
 
+  @State() computedStep: number;
+
+  @Watch("type")
+  @Watch("step")
+  setComputedStep(): void {
+    const { step, type } = this;
+
+    this.computedStep = step ? Math.abs(step) : type === "number" ? 1 : null;
```

Comment:
☝️ `step` supports floats in addition to integers, so we probably shouldn't do `Math.abs` here.  This is particularly relevant for making sure up and down arrows increment properly when decimals are expected in the input and the user wants a finer-grained control over the stepped values.